### PR TITLE
LPD-39807 + LPD-39817

### PR DESCRIPTION
**Pull request by rijema**
Related issues: 

- https://liferay.atlassian.net/browse/LPD-39807
- https://liferay.atlassian.net/browse/LPD-39817

### CODE REVIEW

**Comment by rijema** [comment]
ci:test:sf

File: modules/apps/data-engine/data-engine-rest-test/src/testIntegration/java/com/liferay/data/engine/rest/resource/v2_0/test/DataDefinitionResourceTest.java [hunk 1]
```diff
@@ -125,6 +125,43 @@ public void testDeleteDataDefinition() throws Exception {
...
+					externalReferenceCode, dataDefinition);
+
+		assertHttpResponseStatusCode(
+			200,
+			dataDefinitionResource.
+				getSiteDataDefinitionByContentTypeByExternalReferenceCodeHttpResponse(
+					testGroup.getGroupId(), _CONTENT_TYPE,
+					dataDefinition.getExternalReferenceCode()));
+
+		return dataDefinition;
```
Comment (marcelabc) [inline modules/apps/data-engine/data-engine-rest-test/src/testIntegration/java/com/liferay/data/engine/rest/resource/v2_0/test/DataDefinitionResourceTest.java:162]: Return the method `_addDataDefinition("data-definition-1.json");`

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java [hunk 2]
```diff
@@ -1220,10 +1303,22 @@ private DataDefinition _postSiteDataDefinitionByContentType(
 	}
 
 	private DataDefinition _putDataDefinition(
-			Long dataDefinitionId, DataDefinition dataDefinition,
-			DDMStructure ddmStructure)
+			DataDefinition dataDefinition, DDMStructure ddmStructure)
```
Comment (marcelabc) [inline modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java:1306]: Please do not refactor the existing code.

File: modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java [hunk 3]
```diff
@@ -1828,26 +1926,27 @@ private void _uniquifyDataDefinitionFields(
 	}
 
 	private DataDefinition _updateDataDefinition(
-			DataDefinition dataDefinition, Long dataDefinitionId,
-			DDMForm ddmForm)
+			DataDefinition dataDefinition, DDMStructure ddmStructure)
```
Comment (marcelabc) [inline modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java:1929]: Please do not refactor the existing code (in this situation).

**Pull request closed without merging**